ts/store/middlewares/navigationHistory.ts

```diff
@@ -62,8 +80,19 @@ export function createNavigationHistoryMiddleware(): Middleware<
         // Get the current navigation history
         const currentNavigationHistory = store.getState().navigationHistory;
 
-        // If the history is empty we just need to return
+        // If the history is empty ask to user to press back again or
+        // check if we have to close the app
         if (currentNavigationHistory.length === 0) {
+          const now = new Date().getTime() as Millisecond;
```

Comment:
This logic need to be fixed.

Reply:
@fpersico thanks for the advice ;)
About warnings (_can't perform a react state update..._) I made several tests and now no warnings seem showing. The reason could be about Toast: when the toast hide (and the app is already closed) it makes some state update. This can be prevent by calling Toast.hide() to ensure all toast are closed when the app is closing